content/blog/enforcing-and-automatically-tagging-aws-resources/index.md

```diff
@@ -0,0 +1,745 @@
+        "user:Project": pulumi.getProject(),
+        "user:Stack": pulumi.getStack(),
+        "user:Cost Center": config.require("costCenter"),
+    },
+});
+```
+
+```typescript
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+// Create an S3 bucket (with tags).
+const config = new pulumi.Config();
+const bucket = new aws.s3.Bucket("my-bucket", {
+    tags: {
+        "user:Project": pulumi.getProject(),
+        "user:Stack": pulumi.getStack(),
+        "user:Cost Center": config.require("costCenter"),
+    },
+});
+```
+
+```python
+import pulumi
+import pulumi_aws as aws
+
+# Create an S3 bucket (with tags).
+config = pulumi.Config()
+bucket = aws.s3.Bucket('my-bucket',
+    tags={
+        'user:Project': pulumi.get_project(),
+        'user:Stack': pulumi.get_stack(),
+        'user:Cost Center': config.require('costCenter'),
+    },
+)
+```
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/go/aws/s3"
+    "github.com/pulumi/pulumi/sdk/go/pulumi"
+    "github.com/pulumi/pulumi/sdk/go/pulumi/config"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        // Create an S3 Bucket (with tags):
+        _, err := s3.NewBucket(ctx, "my-bucket", &s3.BucketArgs{
+            Tags: pulumi.Map({
+                "User:Project": pulumi.String(ctx.Project()),
+                "User:Stack": pulumi.String(ctx.Stack()),
+                "User:Cost Center": pulumi.String(config.Require(ctx, "costCenter")),
+            })
+        })
+        return err
+    }
+}
+```
+
+```csharp
+using Pulumi;
+using Pulumi.Aws.S3;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+class Program {
+    static Task Main() {
+        return Deployment.RunAsync(() => {
+            // Create an S3 Bucket (with tags):
+            var config = new Config();
+            var bucket = new Bucket("my-bucket", new BucketArgs {
+                Tags: new Dictionary<string, string> {
+                    { "User:Project",     Deployment.Instance.ProjectName },
+                    { "User:Stack",       Deployment.Instance.StackName },
+                    { "User:Cost Center", config.Get("costCenter") },
+                },
+            });
+        });
+    }
+}
+```
+
+In this example, we're using the project and stack names from the current project, and requiring an explicit cost center configured using `pulumi config set costCenter 11223344`. Of course, this is just for illustration purposes &mdash; we could easily used anything for these keys and values.
+
+Suppose our team requires certain tags, though: How do we ensure we don't forget them?
+
+## Enforcing Tagging Strategies
+
+Policy as Code is a way to enforce infrastructure policies, such as ensuring we're not opening databases to the internet, that we're leveraging strong encryption in all the right places, that we aren't violating cost policies, and so on. And it's a great way to ensure we're tagging everything.
```

Comment:
One general question I have: When should we say "CrossGuard" vs. "Policy as Code"?

Reply:
CrossGuard is a codename, so generally, I've been preferring Policy as Code, especially when used as a peer to Infrastructure as Code, like in this post.

Reply:
Also, in this post, I'm trying to avoid it being very Pulumi-centric and seemingly more general-purpose, just to broaden its appeal (too much marketing speak and codenames seems like it would be at odds with this). That's why the name Pulumi doesn't show up much until the end.